crates/webidl/src/first_pass.rs

```diff
@@ -270,12 +325,12 @@ enum FirstPassOperationType {
     Namespace,
 }
 
-fn first_pass_operation<'src>(
+fn first_pass_operation<'src, A: Into<Arg<'src>> + 'src>(
     record: &mut FirstPassRecord<'src>,
     first_pass_operation_type: FirstPassOperationType,
     self_name: &'src str,
     ids: &[OperationId<'src>],
-    arguments: &'src [Argument<'src>],
+    arguments: impl IntoIterator<Item = A>,
```

Comment:
this change is to allow the generation of operations using `Arg` directly (no instance of `Argument` is available because they are never directly parsed, so its much easier to create the `Arg` directly)